src/core/server/ui_settings/routes/get.ts

```diff
@@ -6,17 +6,28 @@
  * Public License, v 1.
  */
 
+import { schema } from '@kbn/config-schema';
 import { IRouter } from '../../http';
 import { SavedObjectsErrorHelpers } from '../../saved_objects';
 
 export function registerGetRoute(router: IRouter) {
   router.get(
-    { path: '/api/kibana/settings', validate: false },
+    {
+      path: '/api/kibana/settings',
+      validate: {
+        query: schema.object({
+          _allRegistered: schema.maybe(schema.boolean()),
```

Comment:
We are exposing this only for test purposes. What do you think about following a similar approach to the Application Usage's functional tests:

1. A [test plugin](https://github.com/elastic/kibana/tree/master/x-pack/test/usage_collection/plugins/application_usage_test) uses core's public APIs, and calls the existing method `uiSettingsClient.getAll()` and saves it in the `window` object. 
2. Then [the mocha tests](https://github.com/elastic/kibana/blob/master/x-pack/test/usage_collection/test_suites/application_usage/index.ts) retrieve that property and validate their content.

Reply:
I agree exposing an API just for testing might not be ideal. I'm curious if anyone else from the team ( @elastic/kibana-core ) has different ideas.

Using the browser might be ok, even an api in the test plugin that exposes those settings might be even better too. Since `api_integration` tests dont open the browser and using an API is more reliable.